### PR TITLE
[otp_ctrl] update ast cfg software allocation

### PR DESCRIPTION
**Pull request by tjaychen**
Builds upon #12640 
Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Review by Jacob-Levy — APPROVED** [review]
LGTM

**Comment by tjaychen** [comment]
> LGTM

hmm looks like some of our old files are going to fail the new whitespace check.
i'll have a look at this in a bit.

File: util/design/gen-otp-mmap.py [hunk 1]
```diff
@@ -85,6 +85,7 @@ def main():
 
         with open(MMAP_TABLE_FILE, 'w') as outfile:
             outfile.write(TABLE_HEADER_COMMENT + otp_mmap.create_mmap_table())
+            outfile.write("\n")
```
Comment (tjaychen) [inline util/design/gen-otp-mmap.py:88]: @msfschaffner i had to add this since the `trailing_whitespace_check` was failing otherwise.

**Review by msfschaffner — APPROVED** [review]

**Comment by Jacob-Levy** [comment]
@tjaychen In the real design the latches are needed (scan_mode),. In the perevious version, the scan_mode_i & scan_reset_ni, 
were local and were assiged to inactive which optimized out  the latches.

Made them local again for the OS, rebase & rerun the draft.

**Comment by tjaychen** [comment]
sounds good, thanks jacob.

**Comment by Jacob-Levy** [comment]
@tjaychen Can you please take my draft PR latest file after I fixed the FPGA issue, thanks!

**Pull request merged**